Source/core/CarbonFootprintCore.js

```diff
@@ -15,14 +15,15 @@ CarbonFootprintCore.prototype.computeFootprint = function(distance) {
 
 CarbonFootprintCore.prototype.footprintToString = function(footprint) {
     var carbonUnit = this.settingsProvider.getCarbonEmissionUnit();
-    if (footprint > 1000 && carbonUnit =='g') {
-        footprint = footprint / 1000;
-        carbonUnit = 'kg';
-    }
-    else if (carbonUnit == 'lbs') {
-        footprint = footprint * 0.00220462;  
-    }
-    footprint = Math.round(footprint);
+  if (footprint < 1 && carbonUnit == 'kg') {
+    footprint = footprint * 1000;
+    carbonUnit = 'g';
+  }
+  else if (footprint < 1 && carbonUnit == 'lbs') {
+    footprint = footprint * 16;
+    carbonUnit = 'oz';
+  }
+    footprint = Math.round(footprint*100)/100;
```

Comment:
if there are 2 routes, one with CO2 emission of 1.50 kg and other with 2.49 kg, both will show CO2 emission of 2 kg if not rounded off to decimals.

The precision can also be decreased to 1 decimal place if you like :)